pkg/function-manager/controller.go

```diff
@@ -105,10 +105,21 @@ func (h *funcEntityHandler) Delete(obj entitystore.Entity) error {
 		return errors.Wrapf(err, "Driver error when deleting a FaaS function")
 	}
 
+	runs, err := getFilteredRuns(h.Store, &e.Name, nil)
```

Comment:
If a run is in `INITIALIZED`/`CREATING` state, should wait for them to be `READY` before deleting them?

Reply:
No, but we should ensure that we get any orphaned functions in the controller sync and delete them appropriately from the FaaS.  We are not currently doing this.